src/nimony/semimport.nim

```diff
@@ -67,7 +68,36 @@ proc importSingleFile(c: var SemContext; f1: ImportedFilename; origin: string;
   else:
     result = c.processedModules[suffix]
   let module = addr c.importedModules.mgetOrPut(result, ImportedModule(path: f2))
-  loadInterface suffix, module.iface, result, c.importTab, c.converters, c.methods, exports, mode
+  let beforeErrors = c.dest.len
+  loadInterface suffix, module.iface, result, c.importTab, c.converters, c.methods, exports, c.dest, mode
+  if c.dest.len > beforeErrors:
```

Comment:
Use a helper proc for this logic and ideally make `loadInterface` return a success indicator. Move the parsing logic into `loadInterface`.